python/graphstorm/model/edge_decoder.py

```diff
@@ -339,6 +354,106 @@ def out_dims(self):
+                                                  out_dim=out_dim,
+                                                  multilabel=multilabel,
+                                                  target_etype=target_etype,
+                                                  dropout=dropout,
+                                                  regression=regression)
+
+    def _init_model(self):
+        """ Init decoder model
+        """
+        self.relu = th.nn.ReLU()
+
+        # [src_emb | dest_emb] @ W -> h_dim
+        # Here we assume the source and destination nodes have the same dimension.
+        self.nn_decoder = nn.Parameter(th.randn(self.h_dim * 2, self.h_dim))
+        # [edge_feat] @ W -> h_dim
+        self.feat_decoder = nn.Parameter(th.randn(self.feat_dim, self.h_dim))
+        # combine output of nn_decoder and feat_decoder
+        self.combine_decoder = nn.Parameter(th.randn(self.h_dim * 2, self.h_dim))
+        self.decoder = nn.Parameter(th.randn(self.h_dim, self.out_dim))
+        self.dropout = nn.Dropout(self.dropout)
+
+
+        nn.init.xavier_uniform_(self.nn_decoder,
+                                gain=nn.init.calculate_gain('relu'))
+        nn.init.xavier_uniform_(self.feat_decoder,
+                                gain=nn.init.calculate_gain('relu'))
+        nn.init.xavier_uniform_(self.combine_decoder,
+                                gain=nn.init.calculate_gain('relu'))
+        nn.init.xavier_uniform_(self.decoder,
+                                gain=nn.init.calculate_gain('relu'))
+        if self.regression:
+            self.regression_head = nn.Linear(self.out_dim, 1, bias=True)
+
+    def _compute_logits(self, g, h):
+        """ Compute forword output
+
+            Parameters
+            ----------
+            g : DGLBlock
+                The minibatch graph
+            h : dict of Tensors
+                The dictionary containing the embeddings
+            Returns
+            -------
+            th.Tensor
+                Output of forward
+        """
+        with g.local_scope():
+            u, v = g.edges(etype=self.target_etype)
+            src_type, _, dest_type = self.target_etype
+            ufeat = h[src_type][u]
+            ifeat = h[dest_type][v]
+            efeat = g.edges[self.target_etype].data[EP_DECODER_EDGE_FEAT]
+
+            # [src_emb | dest_emb] @ W -> h_dim
+            h = th.cat([ufeat, ifeat], dim=1)
+            nn_h = th.matmul(h, self.nn_decoder)
+            nn_h = self.relu(nn_h)
+            nn_h = self.dropout(nn_h)
+            # [edge_feat] @ W -> h_dim
+            feat_h = th.matmul(efeat, self.feat_decoder)
+            feat_h = self.relu(feat_h)
+            feat_h = self.dropout(feat_h)
+            # [nn_h | feat_h] @ W -> h_dim
+            combine_h = th.cat([nn_h, feat_h], dim=1)
+            combine_h = th.matmul(combine_h, self.combine_decoder)
+            combine_h = self.relu(combine_h)
+            out = th.matmul(combine_h, self.decoder)
```

Comment:
why do you not concatenate src node, dst node and edge features and then run MLP?

Reply:
Embeddings of src and dst are from NN network. However edge features are raw features, it would be better to project it into the same embeddings space of src and dst.